### PR TITLE
[install_script] add support for fips mode

### PR DESCRIPTION
**Pull request by vickenty**
### What does this PR do?

Add support for FIPS mode to the install script.

In FIPS mode, install additional proxy package and configure agent to
talk to it instead of the DD app directly.

### Describe how to test/QA your changes

Prepare a `US1-FED` API key for testing.

For each OS in Debian, RHEL:
- make a fresh VM, run
   ```
   DD_API_KEY=$DD_API_KEY DD_AGENT_MAJOR_VERSION=7 DD_SITE=ddog-gov.com \
   bash -c "$(curl https://raw.githubusercontent.com/DataDog/datadog-agent/7.39.x/cmd/agent/install_script.sh)"
   ```
- check that `datadog-agent` package is installed, `datadog-fips-proxy` is not installed
- check that `datadog.yaml` does not contain custom URL configuration (only `site` and `api_key` are configured): 
- on the same VM, run
   ```
   DD_API_KEY=$DD_API_KEY DD_AGENT_MAJOR_VERSION=7 DD_SITE=ddog-gov.com DD_FIPS_MODE=1
   bash -c "$(curl https://raw.githubusercontent.com/DataDog/datadog-agent/7.39.x/cmd/agent/install_script.sh)"
   ```
- check that the installation script produced `Keeping old /etc/datadog-agent/datadog.yaml configuration file` message and the `datadog.yaml` file is unchanged
- check that `datadog-fips-proxy` is installed
- make a fresh VM, run
   ```
   DD_API_KEY=$DD_API_KEY DD_AGENT_MAJOR_VERSION=7 DD_SITE=ddog-gov.com DD_FIPS_MODE=1
   bash -c "$(curl https://raw.githubusercontent.com/DataDog/datadog-agent/7.39.x/cmd/agent/install_script.sh)"
   ```
- check that `datadog-agent` and `datadog-fips-proxy` are installed
- check that `datadog.yaml` contains `dd_url` configuration pointing to the proxy (localhost)
- check that the agent is sending metrics (`datadog.agent.running` is present in the app for the test vm)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by sgnn7 — APPROVED** [review]

File: cmd/agent/install_script.sh [hunk 1]
```diff
@@ -661,13 +684,49 @@ else
...
+  if [ -z "$fips_mode" ]; then
+    if [ "$site" ]; then
+      printf "\033[34m\n* Setting SITE in the $nice_flavor configuration: $config_file\n\033[0m\n"
+      $sudo_cmd sh -c "sed -i 's/# site:.*/site: $site/' $config_file"
+    fi
+    if [ -n "$DD_URL" ]; then
+      printf "\033[34m\n* Setting DD_URL in the $nice_flavor configuration: $config_file\n\033[0m\n"
+      $sudo_cmd sh -c "sed -i 's|# dd_url:.*|dd_url: $DD_URL|' $config_file"
+    fi
+  else
```
Comment (bkabrda) [inline cmd/agent/install_script.sh:697]: Let's print a message here about configuring the agent for the FIPS proxy, something like the messages in the `if` clause of this condition.

**Review by bkabrda — CHANGES_REQUESTED** [review]
Thanks for sending this PR. It looks good, but I left a minor comment to address inline. You will also need to provide a changelog entry for installscript.

**Review by bkabrda — APPROVED** [review]

File: releasenotes-installscript/notes/fips-be6e112db1a0b58d.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+features:
+  - |
+    Add FIPS mode.
+
+    When environment variable `DD_FIPS_MODE` is set, install script
+    installs an additional FIPS proxy package and automatically
+    configures Agent to direct all traffic to the backend via the FIPS
+    proxy.
```
Comment (kayayarai) [inline releasenotes-installscript/notes/fips-be6e112db1a0b58d.yaml:9]:
```suggestion
    When the `DD_FIPS_MODE` environment variable is set, the install script
    installs an additional FIPS proxy package and configures Agent to direct 
    all traffic to the backend through the FIPS proxy.
```

**Review by kayayarai — APPROVED** [review]
Minor wording suggestion for the release notes, but 👍

**Pull request merged**